apps/icinga/src/main.cpp

```diff
@@ -0,0 +1,91 @@
+#include <Arduino.h>
+#include <otaserver.h>
+#include <kgfx.h>
+#include <HTTPClient.h>
+#include <ArduinoJSON.h>
+
+// CONFIGURATION
+// Please make sure you enabled and configured the new API in your Icinga 2 installation by either setting it up manually
+// or use the CLI command for the automatic setup, which also creates a root-user for the API with a random password (icinga2 api setup)
+// The default port of the Icinga 2 API is 5665, so the URL to go with would be like: https://myicingaserver.com:5665
+// Requires the following API permission(s): status/query
+// Read more about Icinga 2 API Setup: https://icinga.com/docs/icinga-2/latest/doc/12-icinga2-api/
+String icingaAPIURL = "https://hostname:5665";
+String icingaAPIUsername = "username";
+String icingaAPIPassword = "password";
+int refreshTimeInSeconds = 60;
+// -- CONFIGURATION
+
+OTAServer otaserver;
+KGFX ui;
+unsigned long lastTime = 0;
+TFT_eSprite textSpr = ui.createSprite(240, 240);
+HTTPClient http;
+JsonDocument json;
+
+void setup() {  
+  Serial.begin(460800);
+  Serial.println("Starting app");
+
+  otaserver.connectWiFi(); // DO NOT EDIT.
+  otaserver.run(); // DO NOT EDIT
+
+  ui.init();
+  ui.clear();
+  textSpr.setColorDepth(8);
+  textSpr.fillSprite(TFT_BLACK);
+}
+
+void loop() {
+  if((WiFi.status() == WL_CONNECTED)) {
+    otaserver.handle(); // DO NOT EDIT
+
+    if (((millis() - lastTime) > refreshTimeInSeconds*1000) || lastTime == (unsigned long)(0)) {
+      lastTime = millis();
+      http.setUserAgent("kublet-icinga");
+      http.setAuthorization(icingaAPIUsername.c_str(), icingaAPIPassword.c_str());
+      http.begin(icingaAPIURL + "/v1/status/CIB");
+      int httpResponseCode = http.GET();
+
+      if (httpResponseCode < 0) {
+        ui.drawText(textSpr, "HTTP Error: Probably incorrect or not-reachable Icinga 2 API URL.", Arial_14_Bold, TFT_RED, 0, 0);
+        return;
+      }
+
+      if (httpResponseCode != HTTP_CODE_OK) {
+        ui.drawText(textSpr, "HTTP Error: Probably using wrong API credentials.", Arial_14_Bold, TFT_RED, 0, 0);
```

Comment:
If auth fails this will continue calling the HTTP endpoint in an infinite loop. If you don't want that, perhaps you can add a check before the HTTP call:

```
String icingaAPIURL = "https://hostname:5665";
String icingaAPIUsername = "username";
String icingaAPIPassword = "password";
int refreshTimeInSeconds = 60;
bool authFailed = false;
....

void loop() {
   if (((millis() - lastTime) > refreshTimeInSeconds*1000) || lastTime == (unsigned long)(0)) {
    if (authFailed) {
      return;
    }
  }
  if (httpResponseCode < 0) {
    ui.drawText(textSpr, "HTTP Error: Probably incorrect or not-reachable Icinga 2 API URL.", Arial_14_Bold, TFT_RED, 0, 0);
    authFailed = true;
    return;
  }
}
```

Or just leave it to the user to fix and let the applet retry every second.